backend/tests/conftest.py

```diff
@@ -11,6 +11,14 @@ def database_url():
     yield
 
 
+@pytest.fixture(autouse=True, scope="function")
```

Comment:
We don't want to clean up database after every test. That's bad practice.
We can have explicit calls to clear some or all tables ; or we could have a complete cleanup at the end of the session. Might be better in `database_url` then ; or called there.

Reply:
hi @rgaudin I have been trying to fix this but couldn't, here's one of the example I tried

```python
@pytest.fixture(scope="function")
async def clear_titles():
    titles = await Title.objects.all()
    for title in titles:
        await title.languages.clear()
        await title.tags.clear()
        await title.metadata.clear()
        await title.delete()
    yield
```

I used this in `test_stub_titles.py`
```python
async def test_titles_endpoint_get_list_single_title(clear_titles, title):
```

but I am getting errors like

```
E               sqlite3.IntegrityError: FOREIGN KEY constraint failed
E               sqlite3.IntegrityError: UNIQUE constraint failed: languages.code
```

Reply:
yes, this is exactly what I've been warning about. Either we have a clean database for every test (what you've been doing) and tests should be fine but you have no idea what's going on or you clean up everything you created upon teardown of each test. That's what we want. The more we wait to do this, the harder it gets as the number of tests grows.

You sample is good albeit you should yield first and then delete stuff as the part up to the yield statement is called on test setup and the part after is called on test teardown.

I've been able to run all your tests by creating three fixtures: `clear_languages`, `clear_titles`, `clear_books` and `clear_book_dict_languages`. I was just trying to validate that it would work.

Just to be clear, here are the rules we want to follow:

- Model fixtures (such as language for instance) should delete themselves.
- If testing a specific model or a model-specific endpoint (language), it's fine to assume we want to clear the table so a `Model.delete(each=True)` teardown is fine.
- If testing a regular general endpoint (like `books/add`), then we want to remove what we created. That would mean a custom teardown fixture for that endpoint.
- We should not have any removal code at the begining of tests.

Based on this, I believe we'd need the following fixtures:

- `clear_languages()`: `Language.delete(each=True)`
- `clear_books()`: `Language.delete(each=True)` for `test_book_create()`
- `clear_titles()`: for `test_title_create()`
- `clear_book_dict()`: removes metadata, tags, languages, title and book created from the `book_dict`

We may bend those rules in the future but ATM, I see no reason to.